modular_skyrat/modules/synths/code/surgery/robot_healing.dm

```diff
@@ -0,0 +1,162 @@
+	implements = list(TOOL_WELDER = 100, /obj/item/stack/cable_coil = 100)
+	repeatable = TRUE
+	time = 25
+
+	/// If this surgery is healing brute damage. Set during operation steps.
+	var/heals_brute = FALSE
+	/// If this surgery is healing burn damage. Set during operation steps.
+	var/heals_burn = FALSE
+	/// How much healing the sugery gives.
+	var/brute_heal_amount = 0
+	/// How much healing the sugery gives.
+	var/burn_heal_amount = 0
+	/// Heals an extra point of damage per X missing damage of type (burn damage for burn healing, brute for brute). Smaller number = more healing!
+	var/missing_health_bonus = 0
+
+/datum/surgery_step/robot_heal/tool_check(mob/user, obj/item/tool)
+	if(implement_type == TOOL_WELDER && !tool.tool_use_check(user, 1))
+		return FALSE
+	return TRUE
+
+/datum/surgery_step/robot_heal/preop(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	var/woundtype
+	if(implement_type == TOOL_WELDER)
+		heals_brute = TRUE
+		heals_burn = FALSE
+		woundtype = "dents"
+	else
+		heals_brute = FALSE
+		heals_burn = TRUE
+		woundtype = "wiring"
+
+	if(istype(surgery,/datum/surgery/robot_healing))
+		var/datum/surgery/robot_healing/the_surgery = surgery
+		if(!the_surgery.surgery_preop_message_sent)
+			display_results(
+				user,
+				target,
+				span_notice("You attempt to fix some of [target]'s [woundtype]."),
+				span_notice("[user] attempts to fix some of [target]'s [woundtype]."),
+				span_notice("[user] attempts to fix some of [target]'s [woundtype]."),
+			)
+
+/datum/surgery_step/robot_heal/initiate(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery, try_to_fail = FALSE)
+	if(..())
+		while((heals_brute && target.getBruteLoss() && tool.tool_use_check(user,1)) || (heals_burn && target.getFireLoss() && tool))
+			if(!..())
+				break
+
+/datum/surgery_step/robot_heal/success(mob/user, mob/living/carbon/target, target_zone, obj/item/tool, datum/surgery/surgery)
+	var/self_message = "You succeed in fixing some of [target]'s damage" //no period, add initial space to "addons"
+	var/other_message = "[user] fixes some of [target]'s damage" //see above
+	var/healed_brute = 0
+
+	if(heals_brute)
+		healed_brute = brute_heal_amount
+		tool.use_tool(target, user, 0, volume=50, amount=1)
+
+	var/healed_burn = 0
+	if(heals_burn)
+		var/obj/item/stack/cable_coil/cable = tool
+		healed_burn = burn_heal_amount
+		if(!cable.amount)
+			return
+		cable.use(1)
+
+	if(missing_health_bonus)
+		if(target.stat != DEAD)
+			healed_brute += round((target.getBruteLoss()/ missing_health_bonus),0.1)
+			healed_burn += round((target.getFireLoss()/ missing_health_bonus),0.1)
+		else //less healing bonus for the dead since they're expected to have lots of damage to begin with (to make TW into defib not TOO simple)
+			healed_brute += round((target.getBruteLoss()/ (missing_health_bonus*5)),0.1)
+			healed_burn += round((target.getFireLoss()/ (missing_health_bonus*5)),0.1)
```

Comment:
Also that `0.1` could probably be a define (and it might already be, although don't quote me on that.
```suggestion
			healed_brute += round((target.getBruteLoss()/ missing_health_bonus), 0.1)
			healed_burn += round((target.getFireLoss()/ missing_health_bonus), 0.1)
		else //less healing bonus for the dead since they're expected to have lots of damage to begin with (to make TW into defib not TOO simple)
			healed_brute += round((target.getBruteLoss() / (missing_health_bonus * 5)), 0.1)
			healed_burn += round((target.getFireLoss() / (missing_health_bonus * 5)), 0.1)
```